viewing_party/party.py

```diff
@@ -1,23 +1,241 @@
+        'watchlist': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+
+    lod_watchlist = user_data["watchlist"]
+    lod_watched = user_data["watched"]
+    # returns [ {movie_dict} , {movie_dict} , ...]
+
+    for movie_dict in lod_watchlist:
+        if movie_dict["title"] == title:
+            # remove from watchlist
+            lod_watchlist.remove(movie_dict)
+            # add to watched
+            lod_watched.append(movie_dict)
+            # exit loop early if found
+            break
 
+    return user_data
+
+
+#wave 2 
+
+def get_watched_avg_rating(user_data):
+    '''
+    return the average rating
+    The average rating of an empty watched list is 0.0
+    user_data = {
+        'watched': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+
+    lod_watched = user_data["watched"]
+    total_rating = 0
+    
+    if lod_watched == []:
+        return 0.0
+    for movie in lod_watched:
+        total_rating += movie["rating"]
+    
+    avg_rating = total_rating/len(lod_watched)
+    return avg_rating
         
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+
+def get_most_watched_genre(user_data):
+    '''
+    return the genre that is the most frequently watched
+    If the value of "watched" is an empty list, get_most_watched_genre should return None.
+    user_data = {
+        'watched': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+
+    lod_watched = user_data["watched"]
+    genre_count = {}
+
+
+    if lod_watched == []:
+        return None
+    for movie in lod_watched:
+        genre = movie["genre"] # assigning value of genre to variable "genre"
+        count = genre_count.get(genre, 0)
+        genre_count[genre] = count + 1
+
+    # Determine the most frequent
+    max_count = 0
+    frequent_genre = None
+
+    for key, value in genre_count.items():
+        if value > max_count:
+            max_count = value
+            frequent_genre = key
+    return frequent_genre
 
 # -----------------------------------------
-# ------------- WAVE 5 --------------------
+# ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    '''
+    Return a list of dictionaries, that represents a list of movies that:
+    -the user has watched, but none of their friends have watched. (user movies) - (friends movies)
+    user_data = {
+        'watched': [{...}, {...}, {...}, {...}, {...}, {...}], 
+        friends': [{'watched': [..{},{},{}..], {'watched': [..{},{},{}..]
+        }
+    '''
+    titles_friends_watched = set()
+
+    for friend_movie_dict in user_data["friends"]:
+        for movie in friend_movie_dict["watched"]:
+            titles_friends_watched.add(movie["title"])

```

Comment:
This function currently has 2 responsibilities: getting all the movies friends have watched and then finding unique movies. Prefer this logic to be pulled into a helper function so that it can be called here.